src/platform/packages/shared/deeplinks/observability/locators/streams.ts

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+import { SerializableRecord } from '@kbn/utility-types';
+
+export const STREAMS_APP_LOCATOR_ID = 'STREAMS_APP_LOCATOR';
+
+export type StreamsAppLocatorParams<TPageState = Record<string, unknown>> = SerializableRecord &
```

Comment:
Why the dance with the generic that's filled in somewhere else? Is it to get this type into the `deeplinks` package without lifting all the types? What is the advantage of that?

This seems a bit dangerous, because someone could use `StreamsAppLocatorParams` directly somewhere and things wouldn't be typed properly anymore.

Reply:
Makes sense, I moved the params type to the locator to avoid generics and strictly set the Enrichment url state for the params, this is now strongly typed for any consumer.